ros/api/common/add_group_filter.py

```diff
@@ -8,8 +8,12 @@
 
 def group_filtered_query(query):
     if get_flag_value(FLAG_INVENTORY_GROUPS):
-        if len(get_host_groups()) >= 1:
+        if len(get_host_groups()) >= 1 and None in request.host_groups:
             query = query.filter(System.groups[0]['id'].astext.in_(get_host_groups()) | (System.groups == '[]'))
+        elif len(get_host_groups()) >= 1 and None not in request.host_groups:
+            query = query.filter(System.groups[0]['id'].astext.in_(get_host_groups()))
+        else:
+            query = query.filter((System.groups == '[]'))
```

Comment:
@upadhyeammit I think we removed this else condition [with this PR](https://github.com/RedHatInsights/ros-backend/pull/362), or is it needed again for any scenario I am missing? Let me know!

Reply:
I pushed better change! Request to review!